agent/grpc-external/services/resource/delete.go

```diff
@@ -24,25 +27,26 @@ func (s *Server) Delete(ctx context.Context, req *pbresource.DeleteRequest) (*pb
 	// TODO(spatel): reg will be used for ACL hooks
 	_ = reg
 
-	if req.Version == "" {
-		// Delete resource regardless of the persisted Version. Hence, strong read
+	versionToDelete := req.Version
+	if versionToDelete == "" {
+		// Delete resource regardless of the stored Version. Hence, strong read
 		// necessary to get latest Version
 		existing, err := s.Backend.Read(ctx, storage.StrongConsistency, req.Id)
 		if err != nil {
-			return nil, err
-		}
-		if err = s.Backend.DeleteCAS(ctx, existing.Id, existing.Version); err != nil {
-			return nil, err
+			if errors.Is(err, storage.ErrNotFound) {
+				// deletes are idempotent so no-op if resource not found
+				return &pbresource.DeleteResponse{}, nil
+			}
+			return nil, fmt.Errorf("failed read: %v", err)
```

Comment:
Sorry to be a pain, but would you mind making the status code explicit here? 🙇🏻‍♂️  (see: https://github.com/hashicorp/consul/pull/16842#discussion_r1157085813)

Reply:
👍 Fixed in https://github.com/hashicorp/consul/pull/16756/commits/24714ca30e26d6ac720f02bdfe4232159f549c2c